presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -0,0 +1,160 @@
+  // and has a chance for the sync to occur after the shrink
+  // polling interval. This would result in double shrinks.
+  //
+  // Therefore we decided on values from the memory.stat file
+  // that are real time statistics. At first we tried to use
+  // the calculation suggested by the kernel team RSS+CACHE(+SWAP)
+  // However we noticed that this value was not closely related to the
+  // value in usage_in_bytes which is used to OOMKill. We then looked
+  // at all of the values in the stat file and decided that
+  // inactive_anon + active_anon moves closest to that of
+  // usage_in_bytes
+  //
+  // NOTE: We do not know if cgroup V2 memory.current is a fuzz
+  // value. It may be better than what we currently use. For
+  // consistency we will match cgroup V1 and change if
+  // necessary.
+  int64_t systemUsedMemoryBytes() override {
+    size_t memAvailable = 0;
+    size_t memTotal = 0;
+    size_t inactiveAnon = 0;
+    size_t activeAnon = 0;
+
+    if (statFile_ != "None") {
+      folly::gen::byLine(statFile_.c_str()) |
+          [&](const folly::StringPiece& line) -> void {
+        inactiveAnon =
+            extractNumericConfigValueWithRegex(line, kInactiveAnonRegex);
+        activeAnon = extractNumericConfigValueWithRegex(line, kActiveAnonRegex);
+      };
+
+      // Unit is in bytes.
+      return inactiveAnon + activeAnon;
+    }
+
+    // Last resort use host machine info.
+    folly::gen::byLine("/proc/meminfo") |
+        [&](const folly::StringPiece& line) -> void {
+      memAvailable =
+          extractNumericConfigValueWithRegex(line, kMemAvailableRegex) * 1024;
+      memTotal =
+          extractNumericConfigValueWithRegex(line, kMemTotalRegex) * 1024;
+    };
+    // Unit is in bytes.
+    return (memAvailable && memTotal) ? memTotal - memAvailable : 0;
+  }
+
+  int64_t mallocBytes() const override {
+    VELOX_UNSUPPORTED();
+  }
+
+  void periodicCb() const override {
+    return;
+  }
+
+  bool heapDumpCb(const std::string& filePath) const override {
+    VELOX_UNSUPPORTED();
+  }
+
+  void removeDumpFile(const std::string& filePath) const override {
+    VELOX_UNSUPPORTED();
+  }
+
+ private:
+  const boost::regex kInactiveAnonRegex{R"!(inactive_anon\s*(\d+)\s*)!"};
+  const boost::regex kActiveAnonRegex{R"!(active_anon\s*(\d+)\s*)!"};
+  const boost::regex kMemAvailableRegex{R"!(MemAvailable:\s*(\d+)\s*kB)!"};
+  const boost::regex kMemTotalRegex{R"!(MemTotal:\s*(\d+)\s*kB)!"};
+  const char* kCgroupV1Path = "/sys/fs/cgroup/memory/memory.stat";
+  const char* kCgroupV2Path = "/sys/fs/cgroup/memory.stat";
+  std::string statFile_;
+
+  size_t extractNumericConfigValueWithRegex(
+      const folly::StringPiece& line,
+      const boost::regex& regex) {
+    boost::cmatch match;
+    if (boost::regex_match(line.data(), match, regex)) {
+      if (match.size() > 1) {
+        std::string numStr(match[1].str());
+        return std::stoul(numStr);
```

Comment:
This is incorrect. This will return a 32-bit integer which will overflow and return a negative return value as int64_t.
It either needs to be `std::stroull` or use the original `folly::to<size_t>(numStr);`.

Reply:
Thanks for the catch! Changing to use `std::stroull`